supyr_struct/field_type_methods/decoders.py

```diff
@@ -0,0 +1,212 @@
+'''
+Decoder functions for all standard FieldTypes.
+
+Decoders are responsible for converting bytes into a python object*
+
+*Not all decoders receive bytes objects.
+FieldTypes that operate on the bit level cant be expected to return
+even byte sized amounts of bits, so they operate differently.
+A FieldTypes parser and decoder simply need to
+be working with the same parameter and return data types.
+'''
+__all__ = [
+    # basic decoders
+    'decode_numeric', 'decode_string', 'no_decode',
+    'decode_big_int', 'decode_bit_int', 'decode_raw_string',
+    # specialized decoders
+    'decode_24bit_numeric', 'decode_decimal', 'decode_bit',
+    'decode_timestamp', 'decode_string_hex',
+
+    # wrapper functions
+    'decoder_wrapper', 
+    ]
+
+from decimal import Decimal
+from struct import unpack
+from time import ctime
+
+from supyr_struct.defs.constants import ATTR_OFFS
+
+
+# This function is for wrapping decoders in functions which properly
+# work with FieldTypes where is_block and is_data are both True.
+# This is because the node will be a Block with some attribute
+# that stores the "data" of the node.
+def decoder_wrapper(de):
+    '''
+    '''
```

Comment:
Seems like this comment should be in this docstring.

Reply:
kk